File: SalesDatePrediction/SalesDatePrediction.API/Models/NewOrderDto.cs
namespace SalesDatePrediction.API.Models
{
    public class NewOrderDto
    {
        public int CustomerID { get; set; }
        public int EmployeeID { get; set; }
        public int ShipperID { get; set; }
        public string? ShipName { get; set; }
        public string? ShipAddress { get; set; }
        public string? ShipCity { get; set; }
        public DateTime OrderDate { get; set; }
        public DateTime RequiredDate { get; set; }
        public DateTime ShippedDate { get; set; }
        public decimal Freight { get; set; }
        public string? ShipCountry { get; set; }

        public NewOrderItemDto? Product { get; set; }
    }
}

File: SalesDatePrediction/SalesDatePrediction.Web/src/app/models/NewOrderDto.ts
import { NewOrderItemDto } from "./NewOrderItemDto";

export interface NewOrderDto {
  customerID: string;
  employeeID: number;
  shipperID: number;
  shipName: string;
  shipAddress: string;
  shipCity: string;
  orderDate: string;        
  requiredDate: string;     
  shippedDate: string;      
  freight: number;
  shipCountry: string;

  product: NewOrderItemDto;
}
